homeassistant/components/totalconnect/binary_sensor.py

```diff
@@ -0,0 +1,101 @@
+"""Interfaces with TotalConnect sensors."""
+import logging
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_DOOR,
+    DEVICE_CLASS_GAS,
+    DEVICE_CLASS_SMOKE,
+    BinarySensorDevice,
+)
+
+from . import DOMAIN as TOTALCONNECT_DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up a sensor for a TotalConnect device."""
+    if discovery_info is None:
+        return
+
+    sensors = []
+
+    client_locations = hass.data[TOTALCONNECT_DOMAIN].client.locations
+
+    for location_id in client_locations:
+        for zone in client_locations[location_id].zones:
+            sensors.append(
+                TotalConnectBinarySensor(zone, location_id, client_locations)
+            )
+    add_entities(sensors)
+
+
+class TotalConnectBinarySensor(BinarySensorDevice):
+    """Represent an TotalConnect zone."""
+
+    def __init__(self, zone_id, location_id, locations):
```

Comment:
`self._is_tampered`, `self._is_on` and `self._is_low_battery` are missing in the constructor, set them to `None`

Reply:
concur